ArjunaJTS/integration/src/test/java/com/arjuna/ats/jta/distributed/TestResource.java

```diff
@@ -121,37 +136,46 @@ public TestResource(String nodeName, boolean b, boolean fatalCommit) {
 	public synchronized int prepare(Xid xid) throws XAException, Error {
 		System.out.println("[" + Thread.currentThread().getName() + "] TestResource (" + serverId + ")      XA_PREPARE [" + xid + "]");
 
-		if (readonly)
-			return XA_RDONLY;
-		else {
-			File dir = new File(System.getProperty("user.dir") + "/distributedjta-tests/TestResource/" + serverId + "/");
-			dir.mkdirs();
-			file = new File(dir, new Uid().fileStringForm() + "_");
-			try {
-				file.createNewFile();
-				final int formatId = xid.getFormatId();
-				final byte[] gtrid = xid.getGlobalTransactionId();
-				final int gtrid_length = gtrid.length;
-				final byte[] bqual = xid.getBranchQualifier();
-				final int bqual_length = bqual.length;
-
-				DataOutputStream fos = new DataOutputStream(new FileOutputStream(file));
-				fos.writeInt(formatId);
-				fos.writeInt(gtrid_length);
-				fos.write(gtrid, 0, gtrid_length);
-				fos.writeInt(bqual_length);
-				fos.write(bqual, 0, bqual_length);
-				fos.flush();
-				fos.close();
-			} catch (IOException e) {
-				throw new XAException(XAException.XAER_RMERR);
+		if (fatalPrepare) {
+			if (fatalError == null) {
+				throw new Error();
+			}
+			this.fatalXid = xid;
+			throw fatalError;
+		} else {
+			if (readonly)
+				return XA_RDONLY;
+			else {
+				File dir = new File(System.getProperty("user.dir") + "/distributedjta-tests/TestResource/" + serverId + "/");
+				dir.mkdirs();
+				file = new File(dir, new Uid().fileStringForm() + "_");
+				try {
+					file.createNewFile();
+					final int formatId = xid.getFormatId();
+					final byte[] gtrid = xid.getGlobalTransactionId();
+					final int gtrid_length = gtrid.length;
+					final byte[] bqual = xid.getBranchQualifier();
+					final int bqual_length = bqual.length;
+
+					DataOutputStream fos = new DataOutputStream(new FileOutputStream(file));
+					fos.writeInt(formatId);
+					fos.writeInt(gtrid_length);
+					fos.write(gtrid, 0, gtrid_length);
+					fos.writeInt(bqual_length);
+					fos.write(bqual, 0, bqual_length);
+					fos.flush();
+					fos.close();
+				} catch (IOException e) {
+					throw new XAException(XAException.XAER_RMERR);
+				}
+				return XA_OK;
 			}
-			return XA_OK;
 		}
 	}
 
+
 	public synchronized void commit(Xid id, boolean onePhase) throws XAException {
-		System.out.println("[" + Thread.currentThread().getName() + "] TestResource (" + serverId + ")      XA_COMMIT  [" + id + "]");
+		System.out.println("[" + Thread.currentThread().getName() + "] TestResource (" + serverId + ")      XA_COMMIT  [" + xid + "]");
```

Comment:
Hi Manuel the id part of the logging should go in the second commit in the pull request rather than this one please. Same with line 191 a little below. It should be `this.fatalXid = id;` in this commit please.

Reply:
Done. Thanks Tom

Reply:
Thank you